src/clientAuth/clientSecretBasic.js

```diff
@@ -6,69 +12,65 @@ class ClientSecretBasic extends ClientAuthnMethod {
   }
 
   /**
-   * @param {*} cis Request class instance
-   * @param {*} requestArgs Request arguments
-   * @param {*} httpArgs HTTP arguments
+   * @param {?ResourceRequest} cis Request class instance
+   * @param {?ClientInfo} cliInfo Client information
+   * @param {?Object<string, string>} requestArgs Request arguments
+   * @param {?Object<string, string>} httpArgs HTTP header arguments
+   * @param {?Object<string, string>} kwargs Other extra arguments
+   * @return {!Object<string, string>} HTTP header arguments
    */
-  construct(cis, cliInfo, requestArgs, httpArgs, kwargs) {
-    cliInfo = cliInfo || null;
-    requestArgs = requestArgs || null;
-    httpArgs = httpArgs || null;
-
-    if (httpArgs == null) {
-      httpArgs = {};
-    }
+  construct(cis, cliInfo = null, requestArgs = null, httpArgs = {}, kwargs) {
+    httpArgs = httpArgs || {};
+    
     let passwd = null;
     if (kwargs) {
-      passwd = kwargs['password'];
+      passwd = kwargs.password;
     } else {
-      if (httpArgs['password']) {
-        passwd = httpArgs['password'];
+      if (httpArgs.password) {
+        passwd = httpArgs.password;
       } else {
-        if (cis['client_secret']) {
-          passwd = cis['client_secret'];
+        if (cis.client_secret) {
+          passwd = cis.client_secret;
         } else {
           passwd = cliInfo.client_secret;
         }
       }
     }
-
+    
     let user = null;
     if (kwargs) {
```

Comment:
`const user = kwargs ? kwargs.user || cliInfo.client_id;`

I'd even suggest:

`const user = kwargs && kwargs.user ? kwargs.user || cliInfo.client_id;`

Use the same method in other places.